### PR TITLE
Keyword recordings update script

### PR DESCRIPTION
**Pull request by RidwanHamid501**
Create a script to update keyword recordings every hour for subscribed to keywords. Addresses issue #113

### CODE REVIEW

File: notifications-pipeline/notify_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        response = requests.post(API_ENDPOINT, json=data, timeout=1000)
+        if response.status_code == 200:
+            print("✅ Topic submitted successfully!")
+        else:
+            print(f"""Error: {response.json().get(
+                'message', 'Unknown error')}""")
+    except requests.exceptions.RequestException as e:
+        print(f"Failed to connect to the API. Error: {e}")
+
+def find_unique_keywords(cursor):
```
Comment (Kurt812) [inline notifications-pipeline/notify_pipeline.py:41]: could we add typehints here?

File: notifications-pipeline/notify_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
+"""Script to update keyword recordings for subscribed to keywords"""
+
+# pylint: disable=E0401
+
+from os import environ as ENV
+import requests
+from dotenv import load_dotenv
+import psycopg2
+from psycopg2.extras import RealDictCursor
+
```
Comment (SurinaCS) [inline notifications-pipeline/notify_pipeline.py:10]: import logging

Comment (SurinaCS) [inline notifications-pipeline/notify_pipeline.py:10]:
```suggestion
import logging
from psycopg2.extras import RealDictCursor
from psycopg2.extensions import cursor

```

File: notifications-pipeline/notify_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    """Submit topic details to the API"""
+    try:
+        response = requests.post(API_ENDPOINT, json=data, timeout=1000)
+        if response.status_code == 200:
+            print("✅ Topic submitted successfully!")
+        else:
+            print(f"""Error: {response.json().get(
+                'message', 'Unknown error')}""")
+    except requests.exceptions.RequestException as e:
+        print(f"Failed to connect to the API. Error: {e}")
```
Comment (SurinaCS) [inline notifications-pipeline/notify_pipeline.py:39]:
```suggestion
           logging.info("✅ Topic submitted successfully!")
        else:
            logging.error(f"""Error: {response.json().get(
                'message', 'Unknown error')}""")
    except requests.exceptions.RequestException as e:
        logging.error(f"Failed to connect to the API. Error: {e}")
```

File: notifications-pipeline/notify_pipeline.py (hunk 1 above)
Comment (SurinaCS) [inline notifications-pipeline/notify_pipeline.py:41]:
```suggestion
def find_unique_keywords(cursor: cursor) -> RealDictRow:
```

File: notifications-pipeline/notify_pipeline.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+def fetch_keyword(cursor, id):
+    """Find keyword name from ID"""
+    cursor.execute(
+        """SELECT keyword
+         FROM keywords
+         WHERE keywords_id = %s""", (id,))
+    keywords = cursor.fetchone()
+    return keywords
+
+def main():
```
Comment (SurinaCS) [inline notifications-pipeline/notify_pipeline.py:58]:
```suggestion
def main() -> None:
```

File: notifications/notify.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
+from os import environ as ENV
+import boto3
+import psycopg2
+import psycopg2.extras
+from dotenv import load_dotenv
```
Comment (Kurt812) [inline notifications/notify.py:5]:
```suggestion
from dotenv import load_dotenv
from psycopg2.extras import RealDictCursor
from psycopg2.extensions import cursor
```

File: notifications/notify.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                    <p>You are receiving this email as a subscriber of Trend Getter. If you no longer wish to receive updates, 
+                    <a href="https://unsubscribe-link.com" style="color: #a3333e;">unsubscribe here</a>.</p>
+                    <p>&copy; 2024 Trend Getter Inc. All rights reserved.</p>
+                </div>
+            </div>
+        </body>
+        </html>
+        """
+        send_email(notification['email'], message)
+        # print(message)
```
Comment (Kurt812) [inline notifications/notify.py:153]:
```suggestion
   
```

File: notifications/notify.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    JOIN keyword_differences kd ON s.keywords_id = kd.keywords_id
+    WHERE s.subscription_status = TRUE
+      AND kd.difference IS NOT NULL
+      AND ABS(kd.difference) >= s.notification_threshold;
+    """
+    cursor.execute(query)
+    return cursor.fetchall()
+
+
+def send_email(email, message):
```
Comment (Kurt812) [inline notifications/notify.py:66]:
```suggestion
def send_email(email: str, message: str) -> None:
```

File: notifications/notify.py [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+                "Charset": CHARSET,
+                "Data": "Trend Getter Update",
+            },
+        },
+        Source="trainee.ridwan.hamid@sigmalabs.co.uk",
+    )
+    print(response)
+
+
+def lambda_handler(event, context):
```
Comment (Kurt812) [inline notifications/notify.py:97]:
```suggestion
def lambda_handler(event: dict[str], context: Context) -> None:
```

**Review by Kurt812 — COMMENTED** [review]
Changes looking good, please commit the suggested typehints

**Review by Kurt812 — APPROVED** [review]
Great work, thanks for the edits

**Pull request merged**